components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultOIDCClaimsCallbackHandler.java

```diff
@@ -0,0 +1,628 @@
+    }
+
+    private Map<String, Object> getClaimsForLocalUser(OAuthAuthzReqMessageContext authzReqMessageContext, Map<String, Object> claims) {
+        try {
+            claims = getClaimsFromUserStore(authzReqMessageContext);
+            if (log.isDebugEnabled()) {
+                log.debug("User attributes not found in cache. Trying to retrieve attribute for user " +
+                        authzReqMessageContext.getAuthorizationReqDTO().getUser());
+            }
+        } catch (UserStoreException | IdentityApplicationManagementException | IdentityException e) {
+            log.error("Error occurred while getting claims for user " +
+                    authzReqMessageContext.getAuthorizationReqDTO().getUser(), e);
+        }
+        return claims;
+    }
+
+    private boolean isFederatedUser(OAuthAuthzReqMessageContext authzReqMessageContext) {
+        return authzReqMessageContext.getAuthorizationReqDTO().getUser().isFederatedUser();
+    }
+
+    private String getAuthorizationCode(OAuthAuthzReqMessageContext authzReqMessageContext) {
+        return (String) authzReqMessageContext.getProperty(ACCESS_TOKEN);
+    }
+
+    /**
+     * Get claims map
+     *
+     * @param userAttributes User Attributes
+     * @return User attribute map
+     */
+    private Map<String, Object> getClaimsMap(Map<ClaimMapping, String> userAttributes) {
+
+        Map<String, Object> claims = new HashMap<>();
+        if (isNotEmpty(userAttributes)) {
+            for (Map.Entry<ClaimMapping, String> entry : userAttributes.entrySet()) {
+                claims.put(entry.getKey().getRemoteClaim().getClaimUri(), entry.getValue());
+            }
+        }
+        return claims;
+    }
+
+    /**
+     * Get claims from user store
+     *
+     * @param requestMsgCtx Token request message context
+     * @return Users claim map
+     * @throws UserStoreException
+     * @throws IdentityApplicationManagementException
+     * @throws IdentityException
+     */
+    private Map<String, Object> getClaimsFromUserStore(OAuthTokenReqMessageContext requestMsgCtx)
+            throws UserStoreException, IdentityApplicationManagementException, IdentityException {
+
+        Map<String, Object> mappedAppClaims = new HashMap<>();
+        String spTenantDomain = getServiceProviderTenantDomain(requestMsgCtx);
+        ServiceProvider serviceProvider =
+                getServiceProvider(requestMsgCtx.getOauth2AccessTokenReqDTO().getClientId(), spTenantDomain);
+
+        if (serviceProvider == null) {
+            return mappedAppClaims;
+        }
+
+        ClaimMapping[] requestedLocalClaimMap = serviceProvider.getClaimConfig().getClaimMappings();
+        if (ArrayUtils.isEmpty(requestedLocalClaimMap)) {
+            return mappedAppClaims;
+        }
+
+        AuthenticatedUser user = requestMsgCtx.getAuthorizedUser();
+        String userTenantDomain = user.getTenantDomain();
+        String username = user.toString();
```

Comment:
Better to construct the username from AuthenticatatedUser rather than depend on toString() method.